auraed/src/cells/cell_service/cell_service.rs

```diff
@@ -237,6 +241,29 @@ impl CellService {
         executables.broadcast_stop().await;
         Ok(())
     }
+
+    #[tracing::instrument(skip(self))]
+    async fn list(&self) -> Result<CellServiceListResponse> {
+        let cells = self.cells.lock().await;
+
+        
+        Ok(CellServiceListResponse { 
+            cells: cells.entries()
+                        .iter()
+                        .map(|cell| 
+                            Cell { 
+                                name: cell.name().to_string(), 
+                                cpu: Some(CpuController { 
+                                    weight: cell.spec().cgroup_spec.cpu.as_ref().and_then(|cpu| cpu.weight.as_ref().map(|w| w.into_inner())), 
+                                    max: cell.spec().cgroup_spec.cpu.as_ref().and_then(|cpu| cpu.max.as_ref().map(|m| m.into_inner())) }), 
+                                cpuset: Some(CpusetController { cpus: Some(String::from("")), mems: Some(String::from("")) }), 
+                                isolate_process: cell.spec().iso_ctl.isolate_process, 
+                                isolate_network: cell.spec().iso_ctl.isolate_network, 
+                            })
+                        .map(|cell| CellWithChildren { cell: Some(cell), children: vec!() })
+                        .collect_vec()
```

Comment:
This just calls `collect` in a fn that has a `Vec` return type, giving the compiler the required hint for what is being collected. But, since the `cells` field is being set here, the compiler should have enough data and `collect` should be enough. Alternatively, this should be equivalent to `.collect::<Vec<_>>()`